sw/device/silicon_creator/lib/drivers/flash_ctrl.c

```diff
@@ -94,7 +90,19 @@ static void transaction_start(transaction_params_t params) {
       bitfield_bit32_read(params.partition, FLASH_CTRL_PARTITION_BIT_IS_INFO);
   const uint32_t info_type = bitfield_field32_read(
       params.partition, FLASH_CTRL_PARTITION_FIELD_INFO_TYPE);
-  const bool bank_erase = params.erase_type == kFlashCtrlEraseTypeBank;
+  bool bank_erase;
```

Comment:
I think the trick is probably to follow the write to the register with a read and then check that the resulting bit matches `params.erase_type`. You'd grab the relevant bit out of the value that you'd read; expand it to `kFlashCtrlEraseTypeBank` or `kFlashCtrlEraseTypePage` and then (pushing the result through `launder32` to avoid compiler cleverness) compare for equality with `params.erase_type`.

Reply:
I think relying on explicit checks on the data is the main countermeasure here but adding a check here wouldn't hurt either, which should also cover the ADDR register. I'd prefer using sec_mmio instead of inlining something here but I don't think we need/want to update the sec_mmio table in this case and incur that overhead. Created #11146.